tests/unit/mesh/test_boundary_factory.cxx

```diff
@@ -0,0 +1,92 @@
+#include "gtest/gtest.h"
+
+#include "boundary_factory.hxx"
+#include "boundary_region.hxx"
+
+#include "test_extras.hxx"
+
+/// Global mesh
+extern Mesh *mesh;
+
+class TestBoundary : public BoundaryOp {
+public:
+  BoundaryOp *clone(BoundaryRegion *UNUSED(region), const std::list<std::string> &args,
+                    const std::map<std::string, std::string> &keywords) override {
+    auto *testboundary = new TestBoundary();
+    testboundary->args = args;
+    testboundary->keywords = keywords;
+    return testboundary;
+  }
+  std::list<std::string> args;
+  std::map<std::string, std::string> keywords;
+
+  void apply(Field2D &UNUSED(f)) override {}
+  void apply(Field3D &UNUSED(f)) override {}
+};
+
+TEST(BoundaryFactoryTests, IsSingleton) {
+  BoundaryFactory* fac1 = BoundaryFactory::getInstance();
+  BoundaryFactory* fac2 = BoundaryFactory::getInstance();
+
+  EXPECT_EQ(fac1, fac2);
+}
+
+TEST(BoundaryFactoryTests, CreateTestBoundary) {
+  mesh = new FakeMesh(3, 3, 3);
+  
+  BoundaryFactory* fac = BoundaryFactory::getInstance();
+  fac->add(new TestBoundary(), "testboundary");
+  
+  BoundaryRegionXIn region("test_region", 0, 1, mesh);
+
+  // Check no brackets
+  
+  auto *boundary = fac->create("testboundary", &region);
+
+  EXPECT_TRUE( boundary != nullptr );
+  
+  EXPECT_EQ(typeid(*boundary), typeid(TestBoundary));
+
+  delete boundary;
```

Comment:
Please could you split this into multiple tests? The first few lines could be a fixture

Reply:
Sorry, me being lazy. Is there a good way to ensure that the boundary factory singleton is deleted?

Reply:
`BoundaryFactory::cleanup` looks like it does the job, although if you put the fixture code into `static void SetUpTestCase` (note spelling!), it'll only get called once for all the tests. I think that's ok in this case, as none of the tests modify the factory